### PR TITLE
Allow to run tests in a container

### PR DESCRIPTION
**Pull request by poncovka**
Use `make container-test` to run pylint checks and unit tests in a container.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,13 +8,6 @@ TESTS = tests \
 
 DEFAULT_INSTALL_OF_PO_FILES ?= yes
 
-OSVERSION := $(shell grep -o " [0-9]\{1,\}" /etc/redhat-release | sed "s/ //g")
-ifeq ($(OSVERSION),7)
-	PYVERSION = ""
```
Comment (poncovka) [inline Makefile:13]: I don't think it is still possible to run tests with Python 2.

File: Makefile [hunk 2]
```diff
@@ -124,16 +117,18 @@ update-pot:
...
-	@PYTHONPATH=. py.test$(PYVERSION) --processes=-1 -vw tests/
+CONTAINER_NAME = oscap-anaconda-addon-ci
+container-test:
+	podman build --tag $(CONTAINER_NAME) --file tests/Dockerfile
+	podman run --volume .:/oscap-anaconda-addon:Z $(CONTAINER_NAME) make test
+
+test: runpylint unittest
 
 runpylint:
-	@find . -name '*.py' -print|xargs -n1 --max-procs=$(NUM_PROCS) pylint$(PYVERSION) -E 2> /dev/null
```
Comment (poncovka) [inline Makefile:135]: I wasn't able to set up PYTHONPATH in this line.

File: org_fedora_oscap/ks/oscap.py [hunk 3]
```diff
@@ -433,6 +435,7 @@ def setup(self, storage, ksdata, payload):
                 else:
                     # Let's sleep forever to prevent any further actions and
                     # wait for the main thread to quit the process.
+                    # pylint: disable-msg=E1101
```
Comment (poncovka) [inline org_fedora_oscap/ks/oscap.py:438]: Pylint for some reason ignores these comments, so there is the top one.

File: Makefile [hunk 4]
```diff
@@ -124,16 +117,18 @@ update-pot:
 install-po-files:
 	$(MAKE) -C po install RPM_BUILD_ROOT=$(DESTDIR)
 
-test:
-	@echo "***Running pylint$(PYVERSION) checks***"
-	@find . -name '*.py' -print|xargs -n1 --max-procs=$(NUM_PROCS) pylint$(PYVERSION) -E 2> /dev/null
-	@echo "[ OK ]"
-	@echo "***Running unittests checks***"
-	@PYTHONPATH=. py.test$(PYVERSION) --processes=-1 -vw tests/
```
Comment (poncovka) [inline Makefile:132]: I wasn't able to find this tool.

Comment (matejak) [reply Makefile:132]: That's OK, `python -m pytest` does the same and even doesn't need the `PYTHONPATH` set.

**Comment by matejak** [comment]
This is a great addition, many thanks!

**Pull request merged**